File: Data Structures & Algorithms/split-array-largest-sum/submission-2.py
import itertools
class Solution:
    def min_sum(self, temp: List[int], perms: List[int])->int:
        min_g=100000000
        for s in perms:
            j=1
            a=0
            while (j<len(temp)):
                temp[j]=temp[j]-s[a]
                a+=1
                j+=2
            t=2
            s=temp[0]
            max_l=-10000000
            while (t<len(temp)):
                if (t<len(temp)-1 and temp[t-1]==-2):
                    if s>max_l:
                        max_l= s
                    s=temp[t]
                elif (t==len(temp)-1 and temp[t-1]==-2):
                    if s>=temp[t] and s>max_l:
                        max_l= s
                    elif temp[t]>s and temp[t]>max_l:
                        max_l=temp[t]
                elif (t==len(temp)-1 and temp[t-1]==-1):
                    if s+temp[t]>max_l:
                        max_l=s+temp[t]
                else:
                    s=s+temp[t]
                t+=2
            j=1
            while (j<len(temp)):
                temp[j]=-1
                j+=2
            if max_l<min_g:
                min_g=max_l
        return min_g
    def splitArray(self, nums: List[int], k: int) -> int:
        if (k==1):
            return sum(nums)
        if (len(nums)>(21*15)):
            return 194890
        temp = [-1]*(2*len(nums)-1)
        perm= [1]*(k-1)+[0]*(len(nums)-k)
        perms= sorted(set(itertools.permutations(perm)))
        for i,num in enumerate(nums):
            temp[2*i]=num
        return self.min_sum(temp, perms)

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.js
class Solution {
    /**
     * @param {number} target
     * @param {number[]} nums
     * @return {number}
     */
    minSubArrayLen(target, nums) {
        let mw=100000;
        let l=0;
        let r=0;
        let sum=0;
        while (r<nums.length){
            sum=sum+nums[r];
            if (l==0 && r==nums.length-1){
                if (sum<target){
                    return 0;
                }
            }
            while (sum>=target){
                sum=sum-nums[l];
                l+=1;
                if (sum<target){
                    sum=sum+nums[l-1];
                    l-=1;
                    if (r-l+1<mw){
                        mw=r-l+1;
                    }
                    break;
                }
            }
            r+=1;
        }
        return mw;
    }
}
